src/schema/alter-table-builder.ts

```diff
@@ -201,7 +215,19 @@ export class AlterTableBuilder implements ColumnAlteringInterface {
     columns: string[],
     targetTable: string,
     targetColumns: string[]
+  ): AlterTableAddForeignKeyConstraintBuilder
+  addForeignKeyConstraint(
+    columns: string[],
+    targetTable: string,
+    targetColumns: string[]
+  ): AlterTableAddForeignKeyConstraintBuilder
+  addForeignKeyConstraint(
+    ...args: [string, string[], string, string[]] | [string[], string, string[]]
   ): AlterTableAddForeignKeyConstraintBuilder {
+    const constraintName = args.length === 4 ? args[0] : undefined
+    const columns = args.length === 4 ? args[1] : args[0]
+    const targetTable = args.length === 4 ? args[2] : args[1]
+    const targetColumns = args.length === 4 ? args[3] : args[2]
```

Comment:
This is painful to read, not sure what can be done here.

Reply:
Maybe `const [columns, targetTable, targetColumns] = args.length === 4 ? args.slice(1) : args`